### DISCUSSION TITLE
[Feature Request] Automate test suite trimming for debugging

### DISCUSSION THREAD

**Discussion by ygongdev** [ember-cli/ember-exam#730] [Ideas]
Sometimes, when a test suite fails in its browser id, it could be caused by a combination of other test suites in the same browser id. 

It would be cool to have an option to re-run the browser id replays and remove any test suite that are determined unnecessary for reproducing test failures, so we are left with the minimum amount of tests needed to reproduce the failure.

**Comment by choheekim** [comment]
@step2yeung has worked on `test bisecting` to find a smallest set of test modules to reproduce test failure. @step2yeung, could you be able to provide plans/status on the feature?

**Comment by step2yeung** [comment]
So far, there hasn't been planned work to take the proof of concept into full implementation.

**Comment by MichalBryxi** [comment]
Putting it here just in case there it is mathematically not completely stupid idea. Just pseudo-code, but I hope it captures the idea.

```js
let all_tests = [...]; // each test is given unique ID
let X = ...; // order number of offending test, user will have to provide this (?)
let offending_tests = all_tests; // at start every test can be the problematic one
let N = ...; // number of iterations to go though, user will have to provide this (?)

for(var i=0; i < N; i++){ // repeat only N times
  exam.runWithArgs(`--random=${i}`); // run ember exam with new seed each time
  if(exam.testFailed(X)) { // if the test we're interested in failed
    let failed_after = exam.failedAfter(X); // give me the list of tests that ran *after* my test
    offending_tests = array_substract(offending_tests, failed_after); // those tests won't be the problematic ones
  }
}
```
